vassal-app/src/main/java/VASSAL/counters/GlobalCommandTarget.java

```diff
@@ -244,6 +249,90 @@ public boolean equals(Object o) {
     return encode().equals(t.encode());
   }
 
+/*
+  protected boolean fastMatchLocation = false; // True if we are doing Fast Match by location (else other values in this block unused)
+  protected Target targetType = Target.MAP;    // Type of location Fast Match we are doing
+  protected Expression targetMap;              // Specified Map (for MAP, ZONE, LOCATION, XY types)
+  protected Expression targetBoard;            // Specified Board (for XY type)
+  protected Expression targetZone;             // Specified Zone (for ZONE type)
+  protected Expression targetLocation;         // Specified Location (for LOCATION type)
+  protected Expression targetDeck;             // Specified Deck (for DECK type)
+  protected Expression targetX;                // Specified X (for XY type)
+  protected Expression targetY;                // Specified Y (for XY type)
+
+  protected boolean fastMatchProperty = false; // True if we're doing a Fast Match by property value (else next two values ignored)
+  protected Expression targetProperty;         // Name/Key of Fast Match property
+  protected Expression targetValue;            // Value to match for that property
+  protected CompareMode targetCompare;         // Comparison mode
+*/
+
+  /**
+   * @return a list of the item's string/expression fields if any (for search)
+   */
+  public List<String> getExpressionList() {
+    final List<String> expList = new ArrayList<>();
+
+    if (fastMatchLocation) {
+      if ((targetType == Target.MAP) || (targetType == Target.ZONE) || (targetType == Target.LOCATION) || (targetType == Target.XY)) {
+        expList.add(targetMap.getExpression());
+      }
+      if (targetType == Target.XY) {
+        expList.add(targetBoard.getExpression());
+      }
+      if (targetType == Target.ZONE) {
```

Comment:
Don't you want these to be `else if` from here down, since these are mutually exclusive with the one above it?

Reply:
True